File: Data Structures & Algorithms/delete-node-in-a-bst/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* deleteNode(TreeNode* root, int key) {
        if (root == nullptr) {
            return root;
        }

        if (key > root->val) {
            root->right = deleteNode(root->right, key);
        } else if (key < root->val) {
            root->left = deleteNode(root->left, key);
        } else {
            if (root->left == nullptr) {
                TreeNode* tmpNode = root->right;
                delete root;
                return tmpNode;
            } else if (root->right == nullptr) {
                TreeNode* tmpNode = root->left;
                delete root;
                return tmpNode;
            } else {
                int minVal = findMinVal(root->right);
                root->val = minVal;
                root->right = deleteNode(root->right, minVal);
            }
        }

        return root;
    }

    int findMinVal(TreeNode* root) {
        while (root != nullptr && root->left != nullptr) {
            root = root->left;
        }

        return root->val;
    }
};

File: Data Structures & Algorithms/sum-root-to-leaf-numbers/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        total = [0]
        currNum = []

        def dfs(node):
            if not node:
                return

            currNum.append(str(node.val))
            if not node.left and not node.right:
                total[0] += int("".join(currNum))
                currNum.pop()
                return
            
            dfs(node.left)
            dfs(node.right)
            currNum.pop()

        dfs(root)
        return total[0]